frontend/src/modules/Settings/Components/Settings.tsx

```diff
@@ -1,24 +1,87 @@
 import { DropdownSelect } from '@core/index'
-import { Box, IconButton, SelectChangeEvent } from '@mui/material'
-import Menu from '@mui/material'
+import { Box, IconButton, SelectChangeEvent, Switch } from '@mui/material'
 import MenuItem from '@mui/material/MenuItem'
 import { useState } from 'react'
 import { Link } from 'react-router-dom'
 import { DASHBOARD_PATH } from '@core/index'
 import { ReactComponent as LogoImg } from '@assets/img/lscicon.svg'
 import { AccountMenu } from 'Dashboard/Components/AccountMenu'
+import { API_ROOT, COURSE_API } from '@core/Constants'
+import { AdministratorsTable } from './AdministratorsTable'
+import { Admin } from './types'
+import axios from 'axios'
 
 export const Settings = () => {
   const [currRoster, setCurrRoster] = useState<string>('SP23')
-  const semesters = ['SU22', 'FA22', 'WI23', 'SP23', 'SU23']
-
   const changeCurrRoster = (event: SelectChangeEvent) => {
     // function to only open the survey of the semester
     setCurrRoster(event.target.value)
   }
+
+  const [semesters, setSemesters] = useState<string[]>([])
+  function getAllSemesters() {
+    axios
+      .get(`${API_ROOT}${COURSE_API}/semester/all`)
+      .then((res) => setSemesters(res.data))
+  }
+  getAllSemesters()
```

Comment:
I believe this should be wrapped in a use Effect hook. Otherwise, it will continuously call this function anytime this component is mounted/rendered.